ksql-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java

```diff
@@ -315,19 +324,36 @@ private void configureNumActiveQueriesForGivenState(
     // legacy
     configureGaugeForState(
         metrics,
-        ksqlServiceId + metricGroupName  + "-" + name,
+        ksqlServiceId + metricGroupName + "-" + name,
         metricGroupName,
+        Collections.emptyMap(),
         state
     );
     // new
     configureGaugeForState(
         metrics,
         name,
         ksqlServiceId + metricGroupName,
+        customMetricsTags,
         state
     );
   }
 
+  private static Map<String, String> createMetricsTags(final String tags) {
+    try {
+      return tags.equals("")
+          ? Collections.emptyMap()
+          : Splitter.on(";").trimResults().withKeyValueSeparator(":").split(tags);
+    } catch (IllegalArgumentException e) {
+      throw new KsqlException(
+          String.format(
+              "Invalid config value for '%s'. value: %s. reason: %s",
+              KsqlConfig.KSQL_CUSTOM_METRICS_TAGS,
+              tags,
+              e.getMessage()));
```

Comment:
Wouldn't be better to catch this error early in the `KsqlContext`?

I feel we should pass the tags to the `KsqlEngineMetrics` already parsed into a Map. Also, it makes sense to have one place where the `KsqlConfig.KSQL_CUSTOM_METRICS_TAGS` is used instead of having 1 in KsqlContext to get the tags, and another in the metrics to throw the exception.

Better yet, would it make sense to have a `KsqlConfig` method that parses these key/value pairs? It might be useful in future configs with similar values. Like `Map<String, String> getKeyValues(String key)` or a different name.